File: Examples/ObjcDemo/RixSDKPodDemo/RixEngineObjcDemo/UI/Rixengine/AlxMainVC.m
#import "AlxMainVC.h"
#import "AlxBannerVC.h"
#import "AlxBannerXibVC.h"
#import "AlxRewardVideoVC.h"
#import "AlxInterstitialVC.h"
#import "AlxInterstitialBannerVC.h"
#import "AlxNativeVC.h"

@implementation AlxMainVC

- (void)viewDidLoad {
    [super viewDidLoad];
    self.navigationItem.title = NSLocalizedString(@"Alx_ad", @"");
}

- (MenuAppearance)menuAppearance {
    return MenuAppearanceCard;
}

- (NSArray<MenuItem *> *)menuItems {
    return @[
        [[MenuItem alloc] initWithTitle:NSLocalizedString(@"banner_ad", @"") makeVC:^UIViewController *{ return [[AlxBannerVC alloc] init]; }],
        [[MenuItem alloc] initWithTitle:NSLocalizedString(@"banner_ad_xib", @"") makeVC:^UIViewController *{ return [[AlxBannerXibVC alloc] init]; }],
        [[MenuItem alloc] initWithTitle:NSLocalizedString(@"rewardVideo_ad", @"") makeVC:^UIViewController *{ return [[AlxRewardVideoVC alloc] init]; }],
        [[MenuItem alloc] initWithTitle:NSLocalizedString(@"interstitial_video_ad", @"") makeVC:^UIViewController *{ return [[AlxInterstitialVC alloc] init]; }],
        [[MenuItem alloc] initWithTitle:NSLocalizedString(@"interstitial_banner_ad", @"") makeVC:^UIViewController *{ return [[AlxInterstitialBannerVC alloc] init]; }],
        [[MenuItem alloc] initWithTitle:NSLocalizedString(@"native_ad", @"") makeVC:^UIViewController *{ return [[AlxNativeVC alloc] init]; }]
    ];
}

- (NSString *)menuSubtitleAtIndex:(NSInteger)index {
    NSArray *subtitles = @[
        @"Flexible formats at the top, middle or bottom of your app.",
        @"Load banner ads using Interface Builder (Xib).",
        @"Users engage with a video ad in exchange for in-app rewards.",
        @"Full-screen video ads at natural breaks or transition points.",
        @"Full-screen banner ads at natural breaks or transition points.",
        @"Ads that match the look and feel of your app."
    ];
    if (index >= 0 && index < subtitles.count) {
        return subtitles[index];
    }
    return nil;
}

@end

File: AlxAdsDemo/UI/Rixengine/AlxMainVC.swift
//
//  AlxMainVC.swift
//  AlxDemo
//
//  Created by liu weile on 2025/3/31.
//

import UIKit

class AlxMainVC: BaseMenuVC {

    override var menuItems: [MenuItem] {[
        (title: NSLocalizedString("banner_ad",             comment: ""), makeVC: { AlxBannerVC() }),
        (title: NSLocalizedString("banner_ad_xib",         comment: ""), makeVC: { AlxBannerXibVC() }),
        (title: NSLocalizedString("rewardVideo_ad",        comment: ""), makeVC: { AlxRewardVideoVC() }),
        (title: NSLocalizedString("interstitial_video_ad", comment: ""), makeVC: { AlxInterstitialVC() }),
        (title: NSLocalizedString("interstitial_banner_ad",comment: ""), makeVC: { AlxInterstitialBannerVC() }),
        (title: NSLocalizedString("native_ad",             comment: ""), makeVC: { AlxNativeVC() }),
    ]}

    override func viewDidLoad() {
        super.viewDidLoad()
        navigationItem.title = NSLocalizedString("Alx_ad", comment: "")
    }

    override var menuAppearance: MenuAppearance { .card }

    override func menuSubtitle(at index: Int) -> String? {
        let subtitles = [
            "Flexible formats at the top, middle or bottom of your app.",
            "Load banner ads using Interface Builder (Xib).",
            "Users engage with a video ad in exchange for in-app rewards.",
            "Full-screen video ads at natural breaks or transition points.",
            "Full-screen banner ads at natural breaks or transition points.",
            "Ads that match the look and feel of your app."
        ]
        guard index >= 0, index < subtitles.count else { return nil }
        return subtitles[index]
    }
}
